### ISSUE TITLE
Logging and profiling support

### ISSUE THREAD

**Issue by klimov-paul** [yiisoft/yii2-httpclient#2]
Should be provided logging and profiling for the sending HTTP requests.
Both features should be configurable, allowing to be disabled in order to save performance.

**Comment by klimov-paul** [comment]
One thing in particular to think about: should default logging and profiling settings be bound to YII_DEBUG contant?

```php
class Client ....
{
    public $enableProfiling = Yii_DEBUG;
}
```

In this way profilling will be automatically enabled in debug mode and disabled in production mode.

**Comment by klimov-paul** [comment]
Resolved by commit 04a4d47
